scripts/test_screeps_rl_experiment_card.py

```
@@ -70,6 +71,198 @@ def test_generated_card_is_training_runner_valid(self) -> None:
+                incumbent_id: runner.StrategyVariant(id=incumbent_id, parameters=incumbent_parameters),
+                territory_id: runner.StrategyVariant(
+                    id=territory_id,
+                    parameters=dict(card_helper.CONSTRUCTION_PRIORITY_FALLBACK_DEFAULTS[territory_id]),
+                ),
+            }
+
+            with mock.patch.object(runner, "load_strategy_registry", return_value=registry):
+                with self.assertRaisesRegex(
+                    card_helper.CardValidationError,
+                    "variant construction-priority\\.incumbent\\.v1 must define finite construction-priority parameters",
+                ):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)